src/newgrf_commons.cpp

```diff
@@ -669,12 +671,12 @@ void SpriteLayoutProcessor::ProcessRegisters(uint8_t resolved_var10, uint32_t re
 			uint8_t var10 = (flags & TLF_SPRITE_VAR10) ? regs->sprite_var10 : (ground && this->separate_ground ? 1 : 0);
 			if (var10 == resolved_var10) {
 				/* Apply registers */
-				if ((flags & TLF_DODRAW) && GetRegister(regs->dodraw) == 0) {
+				if ((flags & TLF_DODRAW) && object.GetRegister(regs->dodraw) == 0) {
 					result.image.sprite = 0;
 				} else {
 					if (HasBit(result.image.sprite, SPRITE_MODIFIER_CUSTOM_SPRITE)) result.image.sprite += resolved_sprite;
 					if (flags & TLF_SPRITE) {
-						int16_t offset = (int16_t)GetRegister(regs->sprite); // mask to 16 bits to avoid trouble
+						int16_t offset = (int16_t)object.GetRegister(regs->sprite); // mask to 16 bits to avoid trouble
```

Comment:
```suggestion
						int16_t offset = static_cast<int16_t>(object.GetRegister(regs->sprite)); // mask to 16 bits to avoid trouble
```
Though `SpriteID` is now 32 bits, so... might more than 64k sprites be possible in NewGRFs now?

Reply:
This is just an offset inside a sprite set. It's actually range-checked against a `uint16_t` in the next line.